gobblin-cluster/src/main/java/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -149,15 +167,109 @@ public GobblinClusterManager(String clusterName, String applicationId, Config co
   }
 
   /**
+   * Handle leadership change.
+   * The applicationLauncher is only started on the leader.
+   * The leader cleans up existing jobs before starting the applicationLauncher.
+   * @param changeContext notification context
+   */
+  private void handleLeadershipChange(NotificationContext changeContext) {
+    if (this.helixManager.isLeader()) {
```

Comment:
That is in another process/node, so we can't check that. Helix has taken leadership away though, so that process should start shutting down when it receives the notification.